core/cloudflow-operator/src/main/scala/cloudflow/operator/CloudflowApplication.scala

```diff
@@ -122,30 +122,71 @@ object CloudflowApplication {
   }
 
   object Status {
+    val Unknown          = "Unknown"
+    val Running          = "Running"
+    val Pending          = "Pending"
+    val CrashLoopBackOff = "CrashLoopBackOff"
+
     def apply(
-        app: CloudflowApplication.CR,
-        streamletStatuses: Vector[StreamletStatus] = Vector()
-    ): Status =
+        spec: CloudflowApplication.Spec
+    ): Status = {
+      val streamletStatuses = createStreamletStatuses(spec)
       Status(
-        app.spec.appId,
-        app.spec.appVersion,
-        streamletStatuses
+        spec.appId,
+        spec.appVersion,
+        streamletStatuses,
+        Unknown
       )
-  }
+    }
 
-  case class Status(
-      appId: String,
-      appVersion: String,
-      streamletStatuses: Vector[StreamletStatus]
-  ) {
+    def createStreamletStatuses(spec: CloudflowApplication.Spec) = {
+      // TODO not match on runtime, this is not great for extensibility.
+      // There are some plans to make replicas mandatory in the CR
+      // and to indicate extraPods required in the deployment to prevent the code below specific to runtimes
+      import cloudflow.operator.runner._
+      spec.deployments.map { deployment =>
+        val expectedPodCount = deployment.runtime match {
+          case AkkaRunner.runtime  ⇒ deployment.replicas.getOrElse(AkkaRunner.DefaultReplicas)
+          case SparkRunner.runtime ⇒ deployment.replicas.getOrElse(SparkRunner.DefaultNrOfExecutorInstances) + 1
+          case FlinkRunner.runtime ⇒ deployment.replicas.getOrElse(FlinkRunner.DefaultReplicas) + 1
```

Comment:
I think that we would rather push this to the `Runner` itf rather than fixing it here.

Reply:
It's a temporary fix we'll handle these defaults completely differently soon.